internal/dockercompose/client.go

```diff
@@ -30,17 +31,24 @@ type DockerComposeClient interface {
 
 type cmdDCClient struct {
 	env docker.Env
+	mu  *sync.Mutex
 }
 
 // TODO(dmiller): we might want to make this take a path to the docker-compose config so we don't
 // have to keep passing it in.
 func NewDockerComposeClient(env docker.LocalEnv) DockerComposeClient {
 	return &cmdDCClient{
 		env: docker.Env(env),
+		mu:  &sync.Mutex{},
 	}
 }
 
 func (c *cmdDCClient) Up(ctx context.Context, configPaths []string, serviceName model.TargetName, shouldBuild bool, stdout, stderr io.Writer) error {
+	// docker-compose up is not thread-safe, because network operations are non-atomic. See:
```

Comment:
did you manage to go from repro -> no repro with this change?
this seems safe to try, but just wanted to observe that it might be possible that the unsafe part happens after `docker-compose up -d` exits, in which case this mutex wouldn't provide safety (or, it might introduce enough of a staggering between network creations that it happens to be safe in practice, vs parallel builds starting at the same time which makes it very likely for them to hit network creation at the same time)

Reply:
yes, i was able to repro pretty regularly before this change, but not after this change.

This is the function with the bug:
https://github.com/docker/compose/blob/53d00f76777d111c192b0674eb89f2073da73cf0/compose/network.py#L47

Here's the Tilt output:

```
Starting Tilt (v0.12.0-dev, built 2020-02-11)…
Installing Tilt NodeJS dependencies…
Beginning Tiltfile execution
yarn install v1.21.1
[1/5] Validating package.json...
[2/5] Resolving packages...
Successfully loaded Tiltfile (715.332542ms)
      jupyter │ 
      jupyter │ Initial Build • jupyter
           db │ 
           db │ Initial Build • db
success Already up-to-date.
Done in 1.42s.
           db │ Creating network "analytics_default" with the default driver
      jupyter │ Creating network "analytics_default" with the default driver
Starting Tilt webpack server…
Creating analytics_db_1 ... done
           db │ 
      jupyter │ Building jupyter
Error: Build Failed: command ["docker-compose" "-f" "/home/nick/src/analytics/docker-compose.yaml" "ps" "-q" "db"] failed
```

Does that make sense?